### DISCUSSION TITLE
Solutions for Flatpak to Flatpak communication

### DISCUSSION THREAD

**Discussion by superuser-miguel** [flatpak/xdg-desktop-portal#1261] [Questions] [answered]
I'm asking my communities to provide feedback on applications that are not 1 to 1 feature complete with their flatpak counterparts. For example : KeepassXC uses Firefox Native Messaging to communicate the Passwords etc. So, a Flatpak to Flatpak communication is broken here, but maybe Portals can provide the answer. Other applications like Blender, Krita cannot work together because of the apps not being able to launch other apps. 

Abiword flatpak cannot use system spell checkers for some reason is odd , but that bug exist. 

So is it feasible, possible to patch the flatpak for Abiword to allow for this feature through Portals? Maybe Blender > Krita workflows can continue if they are launched through Portals?

**Comment by TingPing** [comment]
You're combining a few topics here. But regarding spellcheck, are you just talking about user defined words? What is system provided for spellcheck?

**Reply by superuser-miguel** [reply]
I will look into the Abiword issue myself, as that was provided to be by another user. I'll download the application later on and see what I can dig up. 

I understand how this can be looked at as multiple topics, but Portals should be the answer for these, at least I hope Portals can be. I'm trying to guage from my communities their flatpak issues and how Portals can potentially help here. Most issues go unreported, some could be bugs, others could be opportunities for a Portals feature. 

thanks for replying.

**Reply by TingPing** [reply]
I don't believe portals are very relevant to spell check. Like I said the only thing that I imagine the user really needs shared to the host is user defined words.

**Comment by Mikenux** [comment]
For Native Messaging:
- Issue: https://github.com/flatpak/xdg-desktop-portal/issues/655
- PR: https://github.com/flatpak/xdg-desktop-portal/pull/705
- Discussion about an alternative to the PR: https://github.com/flatpak/xdg-desktop-portal/discussions/1191

For Blender and Krita, what is the expected workflow? Launch the other app? Share a file between apps?

**Comment by TingPing** [comment, marked as answer]
> Other applications like Blender, Krita cannot work together because of the apps not being able to launch other apps.

Here is the issue for this: https://github.com/flatpak/xdg-desktop-portal/issues/283

**Reply by superuser-miguel** [reply]
Wow, That's a bit disheartening. I understand, and I was hoping to help with working out these issues. I'm not sure I would be much help at this point.

**Reply by Mikenux** [reply]
There are different cases mentioned in this issue. What would help initially is to know in which case the Krita-Blender workflow falls: co-editing the same file or exchanging a file of different format between apps (the original case described in the issue)?
